File: mvc-arch/src/main/java/org/example/mvc/service/UserService.java
package org.example.mvc.service;

import org.example.mvc.annotation.Inject;
import org.example.mvc.annotation.Service;
import org.example.mvc.repository.UserRepository;

@Service
public class UserService {
    private final UserRepository userRepository;

    @Inject
    protected UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public UserRepository getUserRepository() {
        return userRepository;
    }
}

File: pre-onboarding/Week3/src/main/kotlin/com/example/gameduo/service/UserService.kt
package com.example.gameduo.service

import com.example.gameduo.dto.UserDTO
import com.example.gameduo.entity.User
import com.example.gameduo.repository.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserService(private val userRepository: UserRepository) {
    @Transactional
    internal fun createUser(): UserDTO {
        val userId: Long = userRepository.createUser(User())
        return UserDTO(userId = userId)
    }

    @Transactional
    internal fun getUser(userId: Long): UserDTO {
        val user: User = userRepository.getUser(userId)
        if (user.userId == -1L) return UserDTO(userId = -1, errorMsg = "존재하지 않는 user")
        return UserDTO(totalScore = user.getTotalScore(), bossRaidHistory = user.getBossRaidDTOHistory())
    }
}